ui/components/app/user-preferenced-token-input/user-preferenced-token-input.component.js

```diff
@@ -18,7 +18,7 @@ export default class UserPreferencedTokenInput extends PureComponent {
     return (
       <TokenInput
         {...restProps}
-        showFiat={!useNativeCurrencyAsPrimaryCurrency}
+        showFiat={useNativeCurrencyAsPrimaryCurrency}
```

Comment:
I don't think this fully captures the intent of [the bug](https://github.com/MetaMask/metamask-extension/issues/18953).  I think what we're looking for is [this comment](https://github.com/MetaMask/metamask-extension/issues/18953#issuecomment-1538347876):

"if fiat is selected, you input the amount in fiat and it converts to the token value".

But on this branch when I set the currency preference to Fiat, I'm still entering the token value instead of USD:

https://github.com/MetaMask/metamask-extension/assets/3500406/580b7f75-8261-4f51-9f6e-7f66b13b4f0e

I suspect this is what we want to show:

I prefer "Crypto":
- Input: 123456 $PEPE
- Subtitle: $1 USD

I prefer "Fiat":
- Input: $1 USD
- Subtitle: 123456 $PEPE

Although let's get a precise answer from product.